src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -45,28 +57,30 @@ pal::string_t fx_muxer_t::resolve_fx_dir(const pal::string_t& muxer_dir, runtime
 
         std::vector<pal::string_t> list;
         pal::readdir(fx_dir, &list);
-        fx_ver_t max_specified = specified;
+        fx_ver_t most_compatible = specified;
         for (const auto& version : list)
         {
             trace::verbose(_X("Inspecting version... [%s]"), version.c_str());
             fx_ver_t ver(-1, -1, -1);
             if (!specified.is_prerelease() && fx_ver_t::parse(version, &ver, true) && // true -- only prod. prevents roll forward to prerelease.
-                ver.get_major() == max_specified.get_major() &&
-                ver.get_minor() == max_specified.get_minor())
+                ver.get_major() == specified.get_major() &&
+                ver.get_minor() == specified.get_minor())
             {
-                max_specified = std::max(ver, max_specified);
+                // Pick the greatest production that differs only in patch.
+                most_compatible = std::max(ver, most_compatible);
             }
             if (specified.is_prerelease() && fx_ver_t::parse(version, &ver, false) && // false -- implies both production and prerelease.
                 ver.is_prerelease() && // prevent roll forward to production.
-                ver.get_major() == max_specified.get_major() &&
-                ver.get_minor() == max_specified.get_minor() &&
-                ver.get_patch() == max_specified.get_patch())
+                ver.get_major() == specified.get_major() &&
+                ver.get_minor() == specified.get_minor() &&
+                ver > specified)
```

Comment:
Roll forwards can break even within 1.0.0-rc2 -> 1.0.0-rc2-a as well. If you want to keep it safe, check for patch too.